mne/chpi.py

```diff
@@ -193,70 +204,163 @@ def _get_hpi_info(info):
         raise RuntimeError('Appropriate cHPI information not found in'
                            'raw.info["hpi_meas"], cannot process cHPI')
     hpi_result = info['hpi_results'][-1]
-    hpi_coils = info['hpi_meas'][-1]['hpi_coils']
-    hpi_num = np.array([h['number'] for h in hpi_coils])
-    pos_order = np.searchsorted(hpi_num, hpi_result['order'])
-    hpi_dig = [d for d in info['dig'] if d['kind'] == FIFF.FIFFV_POINT_HPI]
+    hpi_coils = sorted(info['hpi_meas'][-1]['hpi_coils'],
+                       key=lambda x: x['number'])  # ascending (info) order
+    hpi_dig = sorted([d for d in info['dig']
+                      if d['kind'] == FIFF.FIFFV_POINT_HPI],
+                     key=lambda x: x['ident'])  # ascending (dig) order
+    pos_order = hpi_result['order'] - 1  # zero-based indexing, dig->info
+    # hpi_result['dig_points'] are in FIFFV_COORD_UNKNOWN coords...?
+
     # this shouldn't happen, eventually we could add the transforms
     # necessary to put it in head coords
     if not all(d['coord_frame'] == FIFF.FIFFV_COORD_HEAD for d in hpi_dig):
         raise RuntimeError('cHPI coordinate frame incorrect')
+    # Give the user some info
+    logger.info('HPIFIT: %s coils digitized in order %s'
+                % (len(pos_order), ' '.join(str(o + 1) for o in pos_order)))
+    logger.debug('HPIFIT: %s coils accepted: %s'
+                 % (len(hpi_result['used']),
+                    ' '.join(str(h) for h in hpi_result['used'])))
     hpi_rrs = np.array([d['r'] for d in hpi_dig])[pos_order]
+    # errors = 1000 * np.sqrt((hpi_rrs - hpi_rrs_fit) ** 2).sum(axis=1)
+    # logger.debug('HPIFIT errors:  %s'
+    #              % ', '.join('%0.1f' % e for e in errors))
     hpi_freqs = np.array([float(x['coil_freq']) for x in hpi_coils])
     # how cHPI active is indicated in the FIF file
     hpi_sub = info['hpi_subsystem']
     hpi_pick = pick_channels(info['ch_names'], [hpi_sub['event_channel']])[0]
-    hpi_on = np.sum([coil['event_bits'][0] for coil in hpi_sub['hpi_coils']])
+    hpi_on = [coil['event_bits'][0] for coil in hpi_sub['hpi_coils']]
+    # not all HPI coils will actually be used
+    hpi_on = np.array([hpi_on[hc['number'] - 1] for hc in hpi_coils])
+    assert len(hpi_coils) == len(hpi_on)
```

Comment:
This one should *really* never actually be wrong, it's more like an "active comment" than anything else